plugins/processors/port_name/port_name.go

```diff
@@ -97,11 +108,26 @@ func readServices(r io.Reader) sMap {
 
 func (d *PortName) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
 	for _, m := range metrics {
-		portProto, ok := m.GetTag(d.SourceTag)
-		if !ok {
-			// Nonexistent tag
+
+		var portProto string
+		var fromField bool
+
+		if len(d.SourceTag) > 0 {
+			if tag, ok := m.GetTag(d.SourceTag); ok {
+				portProto = string([]byte(tag))
+			}
+		}
+		if len(d.SourceField) > 0 {
+			if field, ok := m.GetField(d.SourceField); ok {
+				portProto = field.(string)
```

Comment:
This will panic when the field is not a string. You should probably detect the type, support integers, and warn on fields that don't match the supported types.